### PR TITLE
[FC-0036] feat: Update ObjectTagView

### PR DESCRIPTION
**Pull request by ChrisChV**
## Description

Updates ObjectTagView Put request to update multiple taxonomies in one request

## Supporting information

- Github issue: https://github.com/openedx/modular-learning/issues/190
- Related PRs:
    - https://github.com/openedx/frontend-app-course-authoring/pull/939
    - https://github.com/openedx/edx-platform/pull/34490
- Internal ticket: [FAL-3645](https://tasks.opencraft.com/browse/FAL-3645)

## Testing instructions

- Follow the testing instructions of https://github.com/openedx/frontend-app-course-authoring/pull/939

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ChrisChV! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by bradenmacdonald** [comment]
@ChrisChV Is this ready for review?

**Comment by ChrisChV** [comment]
@bradenmacdonald Yes, It's ready :+1:

**Comment by bradenmacdonald** [comment]
@ChrisChV It LGTM. I'll let @yusuf-musleh test it first before I merge it though.

Once we merge this and update edx-platform, won't it break tagging until we've updated course-authoring too though? It may have been better to implement this in a backwards compatible way. But I guess it's fine, as long as we coordinate the merges all at the same time.

**Comment by ChrisChV** [comment]
> Once we merge this and update edx-platform, won't it break tagging until we've updated course-authoring too though?

Yes, It is necessary to merge https://github.com/openedx/frontend-app-course-authoring/pull/939 and https://github.com/openedx/edx-platform/pull/34490 at the same time

**Review by yusuf-musleh — APPROVED** [review]
:+1: @ChrisChV Works great and looks good to me too! Just a small nit/question.

- [X] I tested this: Tested as part of https://github.com/openedx/frontend-app-course-authoring/pull/939
- [X] I read through the code
- [ ] ~I checked for accessibility issues~
- [ ] ~Includes documentation~

File: openedx_tagging/core/tagging/rest_api/v1/views.py [hunk 1]
```diff
@@ -511,54 +510,75 @@ def update(self, request, *args, **kwargs) -> Response:
...
+                taxonomy=taxonomy,
+                object_id=object_id,
+            )
+            if not request.user.has_perm(
+                perm,
+                # The obj arg expects a model, but we are passing an object
+                perm_obj,  # type: ignore[arg-type]
+            ):
+                raise PermissionDenied(
+                    f"You do not have permission to change object tags for {str(taxonomy)} or object_id."
```
[the remark this answers is not in the document]
Comment (ChrisChV) [reply openedx_tagging/core/tagging/rest_api/v1/views.py]: Thanks! I improved the message here https://github.com/openedx/openedx-learning/pull/180/commits/018734ee962031882b733392e1551b7286a217d8

**Comment by openedx-webhooks** [comment]
@ChrisChV 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-learning&entry.867055334=https://github.com/openedx/openedx-learning/pull/180&entry.1484655318=https://github.com/ChrisChV&entry.752974735=2024-04-08+21:06&entry.1917517419=2024-04-15+17:38&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by ChrisChV** [comment]
@bradenmacdonald could you create a new tag with the version `0.9.1`?

**Comment by bradenmacdonald** [comment]
@ChrisChV Done!

**Pull request merged**